mmv1/templates/terraform/decoders/app_engine_domain_mapping.go.erb

```diff
@@ -0,0 +1,18 @@
+if v, ok := res["sslSettings"]; ok {
```

Comment:
The current test is failing because `sslSettings` isn't returned by the API. However, on further inspection, I think the logic I suggested is wrong, and it isn't always possible to determine whether a preexisting domain mapping is MANUAL or AUTOMATIC, as a manually configured certificate may also return a value in `pendingManagedCertificateId` (but none of our tests set it up to do so).

I think the logic we want to use instead is:

* if sslSettings is set
    * If sslManagementType is set, return that value
    * Otherwise, call `d.Get("ssl_settings.0.ssl_management_type")` and store that value in sslManagementType (carrying over the old value from state-  we do not have enough information to detect drift)
*  If ssl_settings was not set, call `d.Get("ssl_settings.0.ssl_management_type")`, create sslSettings, and store the retrieved value in sslManagementType

Sorry about the number of iterations here- this one's proving particularly gnarly to get right.

Reply:
No problem :) I believe it's done